tasks/go_deps.py

```diff
@@ -0,0 +1,117 @@
+import datetime
+import os
+from collections.abc import Iterable
+
+from invoke.context import Context
+from invoke.exceptions import Exit
+from invoke.tasks import task
+
+from tasks.build_tags import get_default_build_tags
+from tasks.diff import BINARIES
+from tasks.flavor import AgentFlavor
+from tasks.go import GOARCH_MAPPING, GOOS_MAPPING
+from tasks.libs.common.color import color_message
+from tasks.libs.common.datadog_api import create_gauge, send_metrics
+
+
+def compute_count_metric(
+    ctx: Context,
+    build: str,
+    flavor: AgentFlavor,
+    platform: str,
+    arch: str,
+    entrypoint: str,
+    timestamp: int | None = None,
+    extra_tags: Iterable[str] = (),
+):
+    """
+    Compute a metric representing the number of Go dependencies of the given build/flavor/platform/arch,
+    and one with only dependencies outside of the agent repository.
+    """
+
+    if not timestamp:
+        timestamp = int(datetime.datetime.now(datetime.UTC).timestamp())
+
+    goos, goarch = GOOS_MAPPING[platform], GOARCH_MAPPING[arch]
+    build_tags = get_default_build_tags(build=build, flavor=flavor, platform=platform)
+
+    env = {"GOOS": goos, "GOARCH": goarch, "CGO_ENABLED": "1"}
+    cmd = "go list -f '{{ join .Deps \"\\n\"}}'"
+    with ctx.cd(entrypoint):
+        res = ctx.run(
+            f"{cmd} -tags {','.join(build_tags)}",
+            env=env,
+            hide='out',  # don't hide errors
+        )
+        assert res
+
+    deps = res.stdout.strip().split("\n")
+    count = len(deps)
+    external = sum(1 for dep in deps if not dep.startswith("github.com/DataDog/datadog-agent/"))
+
+    tags = [
+        f"build:{build}",
+        f"flavor:{flavor.name}",
+        f"os:{goos}",
+        f"arch:{goarch}",
+    ]
+    tags.extend(extra_tags)
+
+    metric_count = create_gauge("datadog.agent.go_dependencies.all", timestamp, count, tags=tags)
+    metric_external = create_gauge("datadog.agent.go_dependencies.external", timestamp, external, tags=tags)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Wouldn't it be cleaner to have those strings stored in global var ? In case it changes in the future

Reply:
I considered I didn't need a global since they were used in a single place, but that could be cleaner